### PR TITLE
implement ScheduleDelete preHandle

### LINKED ISSUE #4367: ScheduleDelete `PreTransactionHandler`

### PR DESCRIPTION
**Pull request by kimbor**
Signed-off-by: Kim Rader <kim.rader@swirldslabs.com>

Implement preHandle method for ScheduleDelete

Fixes #4367, #4706

### CODE REVIEW

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java [hunk 1]
```diff
@@ -26,26 +31,65 @@
...
+                return new InvalidTransactionMetadata(
+                        txn,
+                        payer,
+                        INVALID_SIGNATURE); // I'm not sure whether this is the correct code
+            // no need to add adminKey to required keys b/c it's also the payer key so already there
+        }
+
+        final var meta =
+                new SigTransactionMetadataBuilder(keyLookup).txnBody(txn).payerKeyFor(payer);
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java:91]: You could add this line at start of method, in parity with other prehandle logics

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java [hunk 2]
```diff
@@ -26,26 +31,65 @@
...
+        if (payerKeyOrLookupFailure.failed()) {
+            return new InvalidTransactionMetadata(
+                    txn, payer, payerKeyOrLookupFailure.failureReason());
+        } else {
+            var payerKey = payerKeyOrLookupFailure.key();
+            if (!adminKey.get().equals(payerKey))
+                return new InvalidTransactionMetadata(
+                        txn,
+                        payer,
+                        INVALID_SIGNATURE); // I'm not sure whether this is the correct code
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java:85]: I think all the logic to be used should be from [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/order/SigRequirements.java#L1222).  This seems different

File: hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                        OK,
+                        schedulerKey,
+                        List.of());
+    }
+
+    private TransactionBody scheduleDeleteTransaction() {
+        return TransactionBody.newBuilder()
+                .setTransactionID(TransactionID.newBuilder().setAccountID(scheduleDeleter))
+                .setScheduleDelete(
+                        ScheduleDeleteTransactionBody.newBuilder().setScheduleID(scheduleID))
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java:150]: Please also add Parity tests like [this](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCreateHandlerParityTest.java), to port `SigRequiremenetsTest` unit tests related to scheduleDelete

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java (hunk 1 above)
Comment (kimbor) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java]: Done.

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java (hunk 2 above)
Comment (kimbor) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java]: I believe the logic is the same both places. It's a little clearer after the edits that I just checked in.

Comment (kimbor) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java]: What bothers me is that the specs [here](https://github.com/hashgraph/hedera-services/blob/develop/docs/scheduled-transactions/revised-spec.md#the-scheduledelete-transaction) and [here](https://hashgraph.github.io/hedera-protobufs/#schedule_delete.proto) mention additional return codes that are never actually generated, either in the old or refactored way of doing things, afaict. If a schedule is already deleted or executed, you'll get an `INVALID_SCHEDULE_ID` return code, when you should get  `SCHEDULE_ALREADY_DELETED`, `SCHEDULE_ALREADY_EXECUTED` or `SCHEDULE_PENDING_EXPIRATION`.

File: hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java (hunk 3 above)
Comment (kimbor) [reply hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java:129]: I will do this as a separate PR. Created issue #4706

**Review by mhess-swl — COMMENTED** [review]
Looks good! Just missing the parity tests

File: hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java (hunk 3 above)
Comment (kimbor) [reply hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java:129]: Parity tests implemented

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java (hunk 2 above)
Comment (kimbor) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleDeleteHandler.java]: Per Slack convo with @tinker-michaelj , the specs need to be updated to match current behavior. I will create a separate issue for that.

**Review by mhess-swl — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by nathanklick — APPROVED** [review]
Approved with regards to `module-info.java` and `build.gradle.kts` changes only.

**Pull request merged**